tracing-modality/src/common/ingest.rs

```diff
@@ -195,12 +197,16 @@ impl ModalityIngest {
             })
     }
 
-    pub(crate) async fn async_connect(options: Options) -> Result<Self, ConnectError> {
+    pub(crate) async fn async_connect(mut options: Options) -> Result<Self, ConnectError> {
         let url = url::Url::parse(&format!("modality-ingest://{}/", options.server_addr)).unwrap();
         let unauth_client = IngestClient::connect(&url, false)
             .await
             .context("init ingest client")?;
 
+        // TODO(AJM): Does this NEED to be in the global metadata? The old code did this.
+        let run_id = Uuid::new_v4();
+        options.add_metadata("run_id", run_id.to_string());
```

Comment:
Right now it does because it needs to get applied to every timeline. That *might* change in the future, but can't say for sure.